src/Sarif.Driver/DriverUtilities.cs

```diff
@@ -85,5 +88,49 @@ public static bool ReportWhetherOutputFileCanBeCreated(string outputFilePath, bo
         /// </returns>
         public static bool CanCreateOutputFile(string outputFilePath, bool force, IFileSystem fileSystem)
```

Comment:
>CanCreateOutputFile [](start = 27, length = 19)

This code will throw if there are illegal characters in the output file path. Do we want to handle this situation? This code also does not verify that the system can actually write to the location in question. To address these concerns, we'd need to add logic that attempts to generate a zero-length file at the specified location. This code would need to capture relevant exceptions. All of this feedback points out that these helpers are not properly named, currently. They don't actually verify that the file is creatable, they only ensure that no file exists on disk without having permission to overwrite. Open to your thoughts on filing a bug for tail here, in order to unblock this immediate value.

Reply:
I love the idea of adding these checks. I filed #1659, "Perform additional checks in DriverUtilities.CanCreateOutputFile."

---
In reply to: [316396582](https://github.com/microsoft/sarif-sdk/pull/1653#discussion_r316396582) [](ancestors = 316396582)

Reply:
I don't want to rename the methods. The name matches the intended purpose of the method. We can certainly add code to make it do its job better.

---
In reply to: [316407280](https://github.com/microsoft/sarif-sdk/pull/1653#discussion_r316407280) [](ancestors = 316407280,316396582)